packages/viewer/src/components/viewer/viewer.tsx

```diff
@@ -310,8 +311,21 @@ export class Viewer {
     await this.unload();
 
     if (this.commands != null && this.dimensions != null) {
-      this.resource = LoadableResource.fromUrn(urn);
-      await this.connectStreamingClient(this.resource);
+      const loadableResource = LoadableResource.fromUrn(urn);
+
+      const isCurrentlyLoadingResource =
+        this.connectingPromise != null &&
+        this.resource != null &&
+        this.resource.id === loadableResource.id;
+      if (isCurrentlyLoadingResource) {
+        return;
+      }
```

Comment:
Instead of returning eagerly, it's generally easier to understand if the code that should be executed is contained within the condition block.